### ISSUE TITLE
form submission - 500 server error - Defined.php in ClassLoader.php on line 412 & classes.php on line 6705

### ISSUE THREAD

**Issue by Adobedog1** [mautic/mautic#1983]
## Description

I got a fatal error after form submission. 500 Internal Server Error

Warning: include(): Failed opening '/home/mustasheville/public_html/Mautic_1.2.2/vendor/twig/twig/lib/Twig/Node/Expression/Test/Defined.php' for inclusion (include_path='.:/usr/lib/php:/usr/local/lib/php') in /home/mustasheville/public_html/Mautic_1.2.2/vendor/composer/ClassLoader.php on line 412

Fatal error: Class 'Twig_Node_Expression_Test_Defined' not found in /home/mustasheville/public_html/Mautic_1.2.2/app/cache/prod/classes.php on line 6705

**The email the form generated actually went through! and it logged the contact.**

## Steps to reproduce

I submitted an HTML form from inside a slider. http://www.mustasheville.com/

I'm running PHP Version 5.6.23

Let me know what else you need. You guys are doing a great job! I love the new version! looks great!

**Comment by escopecz** [comment]
Could you please check if `vendor/twig/twig/lib/Twig/Node/Expression/Test/Defined.php` exists in your file system?

**Comment by Adobedog1** [comment]
There is no Test folder inside of Expression folder. There is a test.php file in expression.

![screen shot 2016-07-08 at 6 02 30 am](https://cloud.githubusercontent.com/assets/4500174/16685637/033a1522-44d2-11e6-911e-57d42cd47237.png)

I downloaded a fresh copy of version 2 as the one running was an update. Both Expression files are the same.

**Comment by Gustavo-Ruiz** [comment]
Hi, I have the same case, no 'Test' folder but test.php under Expression.

This is the code error I am getting...

PHP Warning:  include(): Failed opening '/.../vendor/twig/twig/lib/Twig/Node/Expression/Test/Defined.php' for inclusion (include_path='.:/opt/alt/php56/usr/share/pear:/opt/alt/php56/usr/share/php') in /.../vendor/composer/ClassLoader.php on line 412
PHP Fatal error:  Class 'Twig_Node_Expression_Test_Defined' not found in /.../app/cache/prod/classes.php on line 6705

Thank you in advance for the help.

**Comment by escopecz** [comment]
Yes, one file is missing from the vendor libraries. I have no explanation how that could happen. If you have a chance to run [composer](https://getcomposer.org/) on your server, install it, `cd` to Mautic's root and execute `composer install` command.

If that's not possible, [download Mautic 1.4.1](https://github.com/mautic/mautic/archive/1.4.1.zip) and copy the `/vendor/twig/twig/lib/Twig/Node/Expression/Test` folder to your Mautic.

Or, if you can, wait for Mautic 2.0.1 which will be released very soon and the upgrade will fix it for you.

**Comment by Adobedog1** [comment]
I installed composer and still got the same error in the same situation. Then I downloaded 1.4.1 and there are no files in Vendor/

**Comment by escopecz** [comment]
Ah, right. I added link to the GH version which doesn't have vendors compiled. I updated the link in previous comment.

**Comment by alanhartless** [comment]
This is an issue with our packaging script removing vendor Test assuming they are unit tests. I've added an exclusion for Twig so this will be fixed in 2.0.1.
